src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/DocumentService.kt

```diff
@@ -155,4 +159,72 @@ class DocumentService(
     } else {
       documentsRepository.findAllByNomsIdAndCategory(nomsId, category)
     }
+
+  fun deleteDocumentsInS3(keys: ArrayList<UUID?>, nomsId: String) {
+    val toDelete = ArrayList<ObjectIdentifier>()
+    for (objKey in keys) {
+      toDelete.add(
+        ObjectIdentifier.builder()
+          .key(objKey.toString())
+          .build(),
+      )
+
+      val request = GetObjectRequest.builder()
+        .bucket(bucketName)
+        .key(objKey.toString())
+        .build()
+
+      if (request == null) {
+        logger.info { "Document object key $objKey not found in S3 for nomsId $nomsId" }
+        continue
+      }
+
+      val copyRequest = CopyObjectRequest.builder()
+        .sourceBucket(bucketName)
+        .sourceKey(objKey.toString())
+        .destinationBucket(bucketName)
+        .destinationKey(objKey.toString() + "_deleted")
+        .build()
+      s3Client.copyObject(copyRequest)
+    }
+
+    val deleteObjectRequest: DeleteObjectsRequest = DeleteObjectsRequest.builder()
+      .bucket(bucketName)
+      .delete(
+        Delete.builder()
+          .objects(toDelete).build(),
+      )
+      .build()
+
+    s3Client.deleteObjects(deleteObjectRequest)
+  }
+
+  fun deleteUploadDocumentByNomisId(nomsId: String, category: DocumentCategory): DocumentsEntity? {
+    if (!DocumentCategory.entries.contains(category)) {
+      throw ValidationException("Given document category ${category.name} does not exists")
+    }
+    val prisoner = findPrisonerByNomsId(nomsId)
+    val documentsEntityList: List<DocumentsEntity> = documentsRepository.findAllByNomsIdAndCategory(prisoner.nomsId, category)
+    if (documentsEntityList.isEmpty()) {
+      throw ResourceNotFoundException("No document exists for category ${category.name} and prisoner id $nomsId ")
+    }
+
+    val keyList = ArrayList<UUID?>()
+    for (documentEntity in documentsEntityList) {
+      keyList.add(documentEntity.originalDocumentKey)
+      keyList.add(documentEntity.pdfDocumentKey)
+    }
+    try {
+      deleteDocumentsInS3(keyList, nomsId)
+    } catch (ex: Exception) {
+      logger.error { ex.message }
+      throw Exception("Failed to delete the documents in S3 for nomsId $nomsId ")
```

Comment:
just rethrow the existing exception, this will lose debug information

Reply:
the error is logged and throw  a readable message!